.goreleaser.yml

```diff
@@ -0,0 +1,27 @@
+project_name: "argocd-vault-plugin"
+
+builds:
+  - binary: "argocd-vault-plugin"
+    flags: "-trimpath"
+    ldflags: >-
+      -s -w
+      -X "github.com/IBM/argocd-vault-plugin/version.Version={{.Version}}"
```

Comment:
It looks like this will do some code generation to set the variables in `version.go` accordingly? If so, cool. Though it's not obvious how the git tag is interpolated as `.Version` - maybe I need to read the docs for this `goreleaser`

Reply:
It's a feature of the golang linker.  https://golang.org/cmd/link/

```
-X importpath.name=value
	Set the value of the string variable in importpath named name to value.
	This is only effective if the variable is declared in the source code either uninitialized
	or initialized to a constant string expression. -X will not work if the initializer makes
	a function call or refers to other variables.
	Note that before Go 1.5 this option took two separate arguments.
```

Reply:
Learned something new, thanks. And I honestly didn't now the delimitter for the Golang templates could be changed so I just used RegExp 🤷 . But like you said it worked out nice since we also have the specific-path-in-placeholder syntax too.

Sorry if I'm being dense but why does it say `version.Version={{.Version}}` and not `version.Version={{.Tag}}`?

Reply:
I'll switch it to `.Tag`.  I didn't know about that one when I wrote this because it wasn't mentioned in their release documentation.  I only found the link above after you asked about it!